src/app/shared/services/popup.service.ts

```diff
@@ -10,24 +10,65 @@ export class PopupService {
     }
 
     new(options: SweetAlertOptions): Observable<SweetAlertResult> {
-        return from(swal(options));
+        return from(swal({...options, ...this.setDefaultOptions(options)}));
```

Comment:
That's wrong. You want to:
1. apply the defaults, then
2. set the options that are being set.

This way, `new()` method is extendible.